src/internal/parser.ts

```diff
@@ -123,36 +129,40 @@ export class Parser {
     T extends MapDocumentNode | ProfileDocumentNode
   >(
     path: string,
-    assertion: (node: unknown) => T,
-    cache: Record<string, T>
+    guard: (node: unknown) => node is T,
+    cache: Record<string, T>,
+    sourceHash: string
   ): Promise<T | undefined> {
     let fileExists = false;
     try {
       fileExists = (await fsp.stat(path)).isFile();
     } catch (e) {
       void e;
     }
-    if (fileExists) {
-      const parsed = assertion(
-        JSON.parse(await fsp.readFile(path, { encoding: 'utf8' }))
-      );
-      cache[path] = parsed;
+    if (!fileExists) {
+      return undefined;
+    }
+    const loaded = JSON.parse(
+      await fsp.readFile(path, { encoding: 'utf8' })
+    ) as unknown;
+    //Check if valid type
+    if (!guard(loaded)) {
+      return undefined;
+    }
+    //Check if checksum match
+    if (loaded.astMetadata.sourceChecksum !== sourceHash) {
+      console.log('CHECK');
```

Comment:
log

Reply:
FIxed in: https://github.com/superfaceai/one-sdk-js/pull/166/commits/120f1b8f355ae6e288b8e2f3aaecdd8b1ca2bb2c